File: UCF HSPT Documents/2007/Solutions/zero.cpp
// Guitar Zero solution
// Written in C++ by Jobby Johns
// UCF 2007 High School Programming Contest

/* The solution is very simple.  Get the number of score changes.  Start the
 * score at 0.  Then, get each score change one at a time and adjust the score
 * accordingly.  Once all the score changes are done, output the result
 * ("Shreddin" if score is positive, "Guitar Zero" otherwise).
 */

#include <iostream>
#include <fstream>

using namespace std;

int main()
	{
	// open the file
	ifstream infile("zero.in", ios::in);
	
	// variables
	int n;
	int songNum = 1;
	char c;
	
	// get the input value for each song and solve (stop if 0)
	while (infile >> n && n != 0)
		{
		// start the score and adjust it based on the input
		int score = 0;
		for (int i = 0; i < n; ++i)
			{
			infile >> c;
			if (c == '+')
				{
				score++;
				}
			else if (c == '-')
				{
				score--;
				}
			}
		
		// output the result
		cout << "Song " << songNum << ": ";
		if (score > 0)
			{
			cout << "Shreddin" << endl;
			}
		else
			{
			cout << "Guitar Zero" << endl;
			}
		
		// increment song number for next possible song
		songNum++;
		}
	
	//close the file and quit
	infile.close();
	return 0;
	}

File: UCF HSPT Team/src/y2007/zero.java
import java.io.*;

/*
 * 2007 UCF High School Programming Contest
 * Java Solution for Guitar Zero
 */

public class zero
{
	public static void main(String args[]) throws Exception
	{
		// Create the input file.
		BufferedReader in = new BufferedReader(new FileReader("zero.in"));
		
		int count = 1;
		while(true)
		{
			// Read in the length of the song.
			int length = Integer.parseInt(in.readLine());
			
			// If it's zero, we've reached the end of the input file, so break out of the loop.
			if(length == 0)
			{
				break;
			}
			
			// Read in the song string.
			String input = in.readLine();
			
			// The score starts off at zero.
			int score = 0;
			
			// Loop through each character in the input string, and increment
			// or decrement the score depending on the character.
			for(int i = 0; i < input.length(); i++)
			{
				if(input.charAt(i) == '+')
				{
					score++;
				}
				else
				{
					score--;
				}
			}
			
			// If the score is greater than zero, they're "shreddin", otherwise
			// they're just a guitar zero.
			if(score > 0)
			{
				System.out.println("Song "+count+": Shreddin");	
			}
			else
			{
				System.out.println("Song "+count+": Guitar Zero");
			}
			
			// Increment the count for the next song.
			count++; 
		}
	}
}